### PR TITLE
[Fix #3394] Add accessibility labels for Wallet screens

### LINKED ISSUE #3394: Add accessibility labels for Wallet screens
### User Story
As an automation engineer I need accessibility labels to be added for specific elements so I could use them in my Appium scripts.

### Description
Accessibility labels are required for building stable mobile test automation solution. Please, add accessibility labels for all elements listed below.

#### Wallet
- total-amount-value-text
- total-amount-currency-text
- options-menu-button
- send-transaction-button
- receive-transaction-button
- transaction-history-button
- eth-asset-value-text
- snt-asset-value-text

#### Wallet menu
- manage-assets-button
- cancel-button

#### Manage assets
- done-button
- checkbox

#### Send transaction
- back-button
- choose-recipient-button
- choose-asset-button
- specify-amount-button
- advanced-button
- transaction-fee-button
- sign-transaction-button

#### Sign transaction later dialog
- sign-transaction-later-text
- no-button
- yes-button

#### Choose recipient menu
- recent-recipients-button
- scan-qr-code-button
- enter-recipient-address-button
- cancel-button

#### Send transaction. Recipient was choosen
- recipient-name-text
- recipient-address-text
- sign-later-button

#### Transaction fee
- gas-limit-input
- gas-price-input
- amount-input
- total-fee-input
- reset-to-default-button
- done-button

#### Choose recipient menu
- back-button
- recipient-address-input

#### Recipients
- back-button
- contact-name-text
- contact-address-text

#### Scan QR code with wallet address
- scan-qr-code-with-wallet-address-text
- cancel-button

#### Sign transaction
- signing-phrase-text
- enter-password-input
- cancel-button
- sign-transaction-button

#### Transaction was sent
- transaction-sent-text
- got-it-button

#### Receive transaction
- back-button
- share-button
- qr-code-image
- address-text
- sent-transaction-request-button

#### New transaction request
- back-button
- contact-name-text
- contact-address-text
- choose-contact-button
- choose-asset-button
- amount-input
- sent-request-button

#### Select contact menu
- recent-recipients-button
- cancel-button

#### Select contact
- back-button
- contact-name-text
- contact-address-text

#### Transaction history
- back-button
- filters-button
- history-button
- unsigned-transactions-button
- transaction-item
   - amount-text
   - currency-text
   - sender-text
   - show-transaction-button

#### Transaction details
- back-button
- amount-text
- currency-text
- sender-name-text
- sender-address-text
- recipient-name-text
- recipient-address-text

#### Transaction details menu
- copy-transaction-hash-button
- open-etherscan-button
- cancel-text

#### Filter transactions
- done-button
- select-all-button
- filter-item
   - filter-name-text
   - checkbox

#### Unsigned transactions
- unsigned-transactions-counter-text
- unsigned-transaction-item
   - amount-text
   - currency-text
   - recipient-name-text
   - recipient-address-text
   - sign-button
   - delete-button
   - show-transaction-button

#### No unsigned transactions
- no-unsigned-transactions-text

#### Delete transaction dialog
- delete-transaction-text
- cancel-button
- confirm-button

### PR DESCRIPTION
**Pull request by foopang**
### Summary:
Fix #3394
Add accessibility labels for Wallet screens

status: ready

### CODE REVIEW

**Comment by lukaszfryc** [comment]
@jeluard could you quickly check if the code looks right? Also, see related pull requests #3528 #3529

File: src/status_im/ui/components/toolbar/view.cljs [hunk 1]
```diff
@@ -72,11 +72,12 @@
...
-  [react/text {:style      (merge styles/item styles/item-text style
-                                  (when disabled? styles/toolbar-text-action-disabled))
-               :on-press   (when-not disabled? handler)
-               :uppercase? uppercase?}
+(defn text-action [{:keys [style handler disabled? uppercase?] :as props} title]
+  [react/text (merge {:style      (merge styles/item styles/item-text style
+                                         (when disabled? styles/toolbar-text-action-disabled))
+                      :on-press   (when-not disabled? handler)
+                      :uppercase? uppercase?}
+                     (dissoc props :style :handler :disabled? :uppercase?))
```
Comment (jeluard) [inline src/status_im/ui/components/toolbar/view.cljs]: Is this only for `accessibility-label`?

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 2]
```diff
@@ -227,12 +258,18 @@
 (defn details-list-row
   ([label value]
    (details-list-row label value nil))
-  ([label value extra-value]
+  ([label value extra-value & [value-accessibility-label extra-value-accessibility-label]]
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: I'd rather have a single label base and have both labels derive from it.

**Review by jeluard — CHANGES_REQUESTED** [review]

**Comment by jeluard** [comment]
@lukaszfryc Looks mostly good!
Is there some structure / logic in those labels? Are reusable components being labeled with some pattern?

**Comment by lukaszfryc** [comment]
@jeluard I want to use `-button` suffix for clickable elements like `send-transaction-button`. Similar `-text` for text,`-input` for input fields. See more in #3394.

Reusable components like list items should follow the same pattern (for instance `transaction-history-item`). It maybe helpful if we include index of item (`transaction-history-item-1`) but it's not necessary.

**Comment by yevh-berdnyk** [comment]
Send transaction -> Choose recipient -> Recent recipients: contact-name-text label is not added
screen shot 2018-03-12 at 3 00 55 pm

Send transaction. Recipient was chosen: recipient-name-text label is added to recipient item instead of the name text
screen shot 2018-03-12 at 4 54 51 pm

The same on New transaction view:
screen shot 2018-03-12 at 4 07 35 pm

Transaction amount and currency texts have no labels on both transaction list and transaction details
screen shot 2018-03-12 at 4 31 33 pm
screen shot 2018-03-12 at 4 36 59 pm
screen shot 2018-03-12 at 4 43 29 pm

After recovering account when sender or recipient name is not displayed for transaction sender-name-text label is pinned to sender address text
screen shot 2018-03-12 at 4 39 32 pm

**Comment by foopang** [comment]
@yevh-berdnyk Thanks for checking! It's mostly fixed now. but **Transaction amount and currency texts have no labels on both transaction list and transaction details** this is not true because those labels were added to child `TextView`, if you expand the parent `TextView`, you will find accessibility labels for child `TextView`.

screenshot from 2018 03 13 08 46 36


@jeluard I've updated for `details-list-row`, second and third function arguments can now pass a vector of props data with value or simple text value.

**Comment by jeluard** [comment]
@yevh-berdnyk What is the tool you are using?

**Comment by lukaszfryc** [comment]
@jeluard it's most probably the Appium Inspector or just Appium Driver itself.

**Comment by jeluard** [comment]
@foopang Please rebase! Hopefully for the last time, will merge right away.

**Review by jeluard — APPROVED** [review]

**Comment by foopang** [comment]
@jeluard rebased.

**Comment by yevh-berdnyk** [comment]
@jeluard , @lukaszfryc yes, I used Appium inspector

**Comment by jeluard** [comment]
@foopang Thanks!

**Pull request merged**